### PR TITLE
Eliminate Layman errors

### LINKED ISSUE #1086: Layman silent errors

### PR DESCRIPTION
**Pull request by jmacura**
The outcome shall be twofold:
* Shapefile uploaded to Layman should be safely added to map, as its name is translated to ASCII characters in similar way like it happens in Layman itself
* Layman errors should not die silently, but appear in the template (added catch() statements)

fixes #1086

### CODE REVIEW

**Comment by jmacura** [comment]
[Save layer to Layman crashes without displaying error when the data are too large](https://trello.com/c/UOsU0e4a/109-save-layer-to-layman-crashes-without-displaying-error-when-the-data-are-too-large)

**Comment by fzadrazil** [comment]
Layer name gets correctly translated now, but when nginx returns an error (eg. Shapefile > 5 MB) user doesn't get this information. Smaller Shp is stored correctly, but Datasources Layman tab keeps loading and never actually loads anything.

**Comment by jmacura** [comment]
image
Would you consider this a sufficient "fix" @fzadrazil ?

**Comment by fzadrazil** [comment]
This is much better. At least the user is now able to see some error. Would
be perfect if we could display it not as a code, but rendered html.

**Comment by jmacura** [comment]
ok, I gonna look for options.

**Comment by FilipLeitner** [comment]
This might be also included in PR. 
https://github.com/hslayers/hslayers-ng/commit/ad9fe4f16a2e0ef89947d1b2a9cf2dd48cbb3907
Tested on 1.24, it caused error not alowing us to save map composition that include vector data.

**Comment by jmacura** [comment]
> 
> 
> This might be also included in PR.
> [ad9fe4f](https://github.com/hslayers/hslayers-ng/commit/ad9fe4f16a2e0ef89947d1b2a9cf2dd48cbb3907)

Integrated, thanks!

**Comment by jmacura** [comment]
image
Satisfied, @fzadrazil ? 😄

**Comment by jmacura** [comment]
[Release 2.4.0](https://trello.com/c/ZPfH6ews/106-release-240)

**Comment by raitisbe** [comment]
> This might be also included in PR.
> [ad9fe4f](https://github.com/hslayers/hslayers-ng/commit/ad9fe4f16a2e0ef89947d1b2a9cf2dd48cbb3907)
> Tested on 1.24, it caused error not alowing us to save map composition that include vector data as such composition would not be sucesfully validated againts json schema.

Probably can use just 'WFS' instead of 'hs.format.WFS'. LaymanWFS was meant to have some distinction while we were pusing features without actual wfs transactions but using layman rest calls. Those days are gone

**Review by raitisbe — APPROVED** [review]

**Comment by raitisbe** [comment]
Should I merge?

**Comment by fzadrazil** [comment]
> image
> Satisfied, @fzadrazil ? 😄

This is perfect, thank you ;)

**Comment by jmacura** [comment]
> 
> 
> Should I merge?

Ready from my side. Thanks @FilipLeitner for the cd2e209 !
